File: persistence/src/main/java/com/comssa/persistence/question/dto/major/response/ResponseMajorQuestionClassCountDto.java
package com.comssa.persistence.question.dto.major.response;

import com.comssa.persistence.question.domain.common.QuestionCategory;
import com.comssa.persistence.question.domain.common.QuestionLevel;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

@NoArgsConstructor
@Getter
public class ResponseMajorQuestionClassCountDto {
	private QuestionCategory questionCategory;
	private QuestionLevel questionLevel;
	private int count;

	@Builder
	public ResponseMajorQuestionClassCountDto(
		QuestionCategory questionCategory,
		QuestionLevel questionLevel,
		int count) {
		this.questionCategory = questionCategory;
		this.questionLevel = questionLevel;
		this.count = count;
	}

	public static ResponseMajorQuestionClassCountDto of(QuestionCategory questionCategory, QuestionLevel questionLevel,
														int count) {
		return ResponseMajorQuestionClassCountDto.builder()
			.questionLevel(questionLevel)
			.questionCategory(questionCategory)
			.count(count)
			.build();
	}
}

File: persistence/src/main/java/com/comssa/persistence/question/dto/major/response/ResponseMajorQuestionClassCountDto.kt
package com.comssa.persistence.question.dto.major.response

import com.comssa.persistence.question.domain.common.QuestionCategory
import com.comssa.persistence.question.domain.common.QuestionLevel

data class ResponseMajorQuestionClassCountDto(
	val questionCategory: QuestionCategory,
	val questionLevel: QuestionLevel,
	val count: Int,
)
